lib/Backend/Commands.hs

```diff
@@ -257,35 +263,38 @@ renameCmd
      , Member (Error RenameResult) r
      )
   => Config -> RenameOptions -> Sem r RenameResult
-renameCmd config (RenameOptions dryRun (QualifiedPath oldBackendNameMb oldPath) (QualifiedPath newBackendNameMb newPath) force) = do
+renameCmd config (RenameOptions dryRun oldQPath@(QualifiedPath oldBackendNameMb _) newQPath@(QualifiedPath newBackendNameMb _) force) = do
   oldBackend <- getBackend config oldBackendNameMb
   newBackend <- getBackend config newBackendNameMb
-  operations <- buildCopyOperations oldBackend newBackend oldPath newPath force
+  operations <- buildCopyOperations oldBackend newBackend oldQPath newQPath force
 
   unless dryRun do
     runCopyOperations newBackend operations
 
   -- we don't want to delete clashed entries if renaming is forced.
   let pathsToDelete =
-        flip filter operations \(CopyOperation old _) ->
-          none (\(CopyOperation _ new) -> old ^. path == new ^. path) operations
+        flip filter operations \(CopyOperation old _ _ _) ->
+          none (\(CopyOperation _ _ new _) -> old ^. path == new ^. path) operations
 
   -- If directory/entry was successfully copied, then we can delete old directory/entry without delete errors.
   unless dryRun do
-    forM_ pathsToDelete \(CopyOperation old _) -> do
+    forM_ pathsToDelete \(CopyOperation old _ _ _) -> do
       let qPath = QualifiedPath oldBackendNameMb (Path.entryPathAsPath (old ^. path))
       void $ catchAndReturn $ deleteCmd config (DeleteOptions dryRun qPath False)
 
   pure $ CPRSuccess $ getOperationPaths <$> operations
 
 data CopyOperation = CopyOperation
   { coOld :: Entry
+  , coOldBackendNameMb :: Maybe BackendName
```

Comment:
I wonder if some of the code could be simplified if we stored a `coOld :: QualifiedPath Entry` instead of `Entry + Maybe BackendName`....

Then, in `getOperationPaths`, to transform a `QualifiedPath Entry` into a `QualifiedPath EntryPath` we could:
* Derive `Functor` for `QualifiedPath`
* And do `view E.path <$> old`

We could derive `Functor` for `QualifiedPath`,

Reply:
Done